content/source/docs/extend/resources/customizing-differences.html.md

```diff
@@ -0,0 +1,60 @@
+---
+layout: "extend"
+page_title: "Resources - Customizing Differences"
+sidebar_current: "docs-extend-resources-customizing-differences"
+description: |-
+  Difference customization within Resources.
+---
+
+# Resources - Customizing Differences
+
+Terraform tracks the state of provisioned resources in its state file. The user passed configuration is compared against what is in the state file. When there is a detected discrepancy the user is presented with the difference of what is configured versus what is in state. Sometimes these scenarios require special handling, this is where the `CustomizeDiff` function is used. It is passed a `*schema.ResourceDiff`, a structure similar to `schema.ResourceData` but lacking most write functions like `Set`, while introducing new functions that work with the difference such as `SetNew`, `SetNewComputed`, and `ForceNew`.
+
+While any function can be provided for difference customization, it is recommended to try and compose the behavior using the [customdiff](https://godoc.org/github.com/hashicorp/terraform/helper/customdiff) helper package. This will allow for a more declarative configuration; however, it should not be overused, so for highly custom requirements, opt for a tailor-made function.
```

Comment:
> While any function can be provided for difference customization, it is recommended to try and compose the behavior using the [customdiff](https://godoc.org/github.com/hashicorp/terraform/helper/customdiff) helper package

Since the customiseDiff method is broken (as it doesn't work if a value's interpolated) - I actually think we shouldn't be recommending customizeDiff as a route (or at least flagging it with this note), since it leads to duplicating the validation effort when that value could be interpolated (as most fields could) - WDYT?

Reply:
I'll add a `~> *NOTE:*` at the top for now since removing it leaves out (potentially) valuable documentation for the limited situations it does work. 🙁